convert2rhel/grub.py

```diff
@@ -572,6 +581,56 @@ def post_ponr_set_efi_configuration():
         _log_critical_error(e.message)
 
 
+def update_grub_after_conversion():
+    """Update GRUB2 images and config after conversion.
+
+    This is mainly a protective measure to prevent issues in case the original distribution GRUB2 tooling
+    generates images that expect different format of a config file. To be on the safe side we
+    rather re-generate the GRUB2 config file and install the GRUB2 image.
+    We opted for doing that only for GRUB2 and not for GRUB Legacy intentionally for the reason of
+    RHEL 6 having transitioned to the Extended Life-cycle Support phase.
+    """
+
+    if systeminfo.system_info.version.major == 6:
+        logger.warning(
+            "Convert2RHEL does not install updated GRUB Legacy bootloader image on RHEL 6. Install the image manually "
+            "by following https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/installation_guide/sect-grub-installing."
+        )
+        return
+
+    utils.RestorableFile(GRUB2_BIOS_CONFIG_FILE).backup()
+    utils.RestorableFile(GRUB2_BIOS_ENV_FILE).backup()
```

Comment:
@SpyTec now this crossed my mind. We are backing up those 2 files here but not restoring them in a case of rollback. Should we add this to `rollback_changes()` function? The `restore()` function has some conditionals to check if the file was backed-up or not, so I think it will be okay adding that there. 

The only thing is that we would probably doing this: 

```python
from convert2rhel.grub import GRUB2_BIOS_CONFIG_FILE, GRUB2_BIOS_ENV_FILE
utils.RestorableFile(GRUB2_BIOS_CONFIG_FILE).restore()
utils.RestorableFile(GRUB2_BIOS_ENV_FILE).restore()
```

Or even 

```python
# convert2rhel/grub.py 

def update_grub_after_conversion():
    ....
    grub2_bios_config_file.backup()
    grub2_bios_env_file.backup()
    ....
    
grub2_bios_config_file = utils.RestorableFile(GRUB2_BIOS_CONFIG_FILE).backup()
grub2_bios_env_file = utils.RestorableFile(GRUB2_BIOS_ENV_FILE).backup()

# convert2rhel/main.py 

def rollback_changes(): 
   ....
   grub.grub2_bios_config_file.restore()
   grub.grub2_bios_env_file.restore()
   ....
```

What do you think? It's better to do the rollback on the `rollback_changes()` function, or add a condition on `update_grub_after_conversion()` that in case of failure, we do a quick rollback of those 2 files?

Reply:
The backup was something Michal suggested in an earlier review and he gave this comment

> I don't think we need to back the files up before the package replacement (and create two new singletons). The reason to back them up is just to give the customer a way to restore them manually post-conversion. Our rollback feature won't take care of that.

So I wouldn't say we need to restore it. But it allows the customer to restore it themselves if they need it

Reply:
Oh, that's right! My bad. It's been some time without reading the comments here and I totally forgot about it hehe.